game/plugin/skills/fishing/src/org/apollo/game/plugin/skills/fishing/FishingSpot.kt

```diff
@@ -77,10 +77,16 @@ enum class FishingSpot(val npc: Int, private val first: Option, private val seco
         /**
          * A [FishingSpot] [Option] that can provide a two different types of fish.
          */
-        private data class Pair(override val tool: FishingTool, val primary: Fish, val secondary: Fish) : Option() {
+        private data class Pair(override val tool: FishingTool, val primary: Fish, val secondary: Fish, val junk: Array<Fish>) : Option() {
             override val level = Math.min(primary.level, secondary.level)
 
             override fun sample(level: Int): Fish {
+                if (junk.isNotEmpty() && rand(100) < JUNK_WEIGHTING) {
```

Comment:
Hmm, we need a unit test for this, but I'm not quite sure about how we go about stubbing the randomness in this case (cc @Major-).